examples/energy_mng/ui/widgets/value_display.slint

```diff
@@ -0,0 +1,102 @@
+// Copyright © SixtyFPS GmbH <info@slint-ui.com>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-commercial
+
+import { Theme } from "../theme.slint";
+
+component ValueDelegate {
+    private property <float> display-value;
+
+    in property <bool> active;
+    in property <string> title <=> i-title.text;
+    in property <string> unit <=> i-unit.text;
+    in property <float> value;
+    in property <bool> alternative-colors;
+
+    HorizontalLayout {  
+        spacing: 15px;
+        Rectangle {  
+            min_width: 1px;
+            background: alternative-colors ? Theme.palette.slint-blue-300 : Theme.palette.limon-green;
+            horizontal-stretch: 0;
+        }
+
+        VerticalLayout {  
+            alignment: center;
+            horizontal-stretch: 1;
+
+            i-title := Text {  
+                color: alternative-colors ? Theme.palette.slint-blue-300 : Theme.palette.limon-green;
+                font-size: Theme.typo.label.size;
+                font-weight: Theme.typo.label.weight;
+            }
+
+            HorizontalLayout {  
+                alignment: start;
+                spacing: 5px;
+
+                Text {  
+                    color: Theme.palette.white;
+                    text: round(display-value * 100) / 100;
+                    font-size: Theme.typo.value.size;
+                    font-weight: Theme.typo.value.weight;
+                    vertical-alignment: center;
+                }
+
+                i-unit := Text {  
+                    y: Theme.spaces.small;
+                    vertical-alignment: center;
+                    color: alternative-colors ? Theme.palette.slint-blue-300 : Theme.palette.limon-green;
+                    font-size: Theme.typo.label-light.size;
+                    font-weight: Theme.typo.label-light.weight;
+                }
+            }
+        }
+    }
+
+    states [  
+        active when active : {
+            display-value: value;
+
+            in {
+                animate display-value { duration: Theme.durations.slow; }
+            }
+        }
+    ]
+}
+
+export struct Value {
```

Comment:
I suggest to rename this to something that includes what is being displayed. Like "EnergyUsage" (and "EnergyUsageDisplay") - if that's what's being displayed. Is that possible, or is this re-used in totally different contexts?

Reply:
It's an float value, unit and title. I think it can be used also as non energy usage value, therefore I like a generic name.